// @fuxishi/vitepress-theme/dist/config.js
import e from "node:crypto";
import { existsSync as t, readdirSync as n } from "node:fs";
import { dirname as r, extname as i, join as a, resolve as o } from "node:path";
//#region .vitepress/config/plugins/demo-preview.ts
var s = "virtual:fx-demo/", c = /* @__PURE__ */ new Map();
function l(t, n = 10) {
	return e.createHash("sha256").update(t).digest("hex").slice(0, n);
}
function u(e) {
	return i(e).slice(1) || "vue";
}
function d(e) {
	return e.split("\\").join("/");
}
function f(e) {
	let t = e.split("\n"), n = t.filter((e) => e.trim()).map((e) => (e.match(/^\s*/) || [""])[0].length), r = n.length ? Math.min(...n) : 0;
	return t.map((e) => e.slice(r)).join("\n");
}
function p(e) {
	e.core.ruler.after("inline", "fx-demo-preview", (e) => {
		let i = e.env, p = process.cwd(), m = (t) => {
			let n = e.tokens.findIndex((e) => e.type === "html_block" && /^<script setup>/.test(e.content.trim()));
			if (n === -1) {
				let n = new e.Token("html_block", "", 0);
				n.content = `<script setup>\n${t}\n<\/script>\n`, e.tokens.splice(0, 0, n);
			} else e.tokens[n].content.includes(t) || (e.tokens[n].content = e.tokens[n].content.replace("<\/script>", `${t}\n<\/script>`));
		}, h = (t, n, r, i) => {
			let a = [], o = new e.Token("html_inline", "", 0);
			o.content = `<template #${t}>`, a.push(o);
			let s = new e.Token("fence", "code", 0);
			s.info = n, s.content = r, i && (s.src = i), a.push(s);
			let c = new e.Token("html_inline", "", 0);
			return c.content = "</template>", a.push(c), a;
		}, g = () => {
			let t = new e.Token("html_inline", "", 0);
			return t.content = "</DemoPreview>", t;
		}, _ = (t, n, r, i) => {
			e.tokens[t].content = `<DemoPreview files="${encodeURIComponent(JSON.stringify(n))}"><${r}/>`, e.tokens.splice(t + 1, 0, ...i, g());
		};
		e.src = e.src.replace(/<DemoPreview\b[^>]*\bdir="([^"]*)"[^>]*?\/?>/g, (t, r) => {
			let i = d(a(p, r)), o = [];
			try {
				o = n(i) || [];
			} catch {
				o = [];
			}
			if (!o.length) return "";
			o.sort((e, t) => e === "index.vue" ? -1 : t === "index.vue" ? 1 : e.localeCompare(t));
			let s = `Fx_${l(i)}`;
			m(`import ${s} from '${i}/index.vue'`);
			let c = e.tokens.findIndex((e) => e.content.includes(t));
			if (c === -1) return "";
			let f = o.flatMap((e) => {
				let t = d(a(i, e));
				return h(e, u(e), `<<< ${t}`, [t]);
			});
			return _(c, o, s, f), "";
		}), e.src = e.src.replace(/<DemoPreview\b[^>]*\bsrc="([^"]*)"[^>]*?\/?>/g, (n, a) => {
			let s = d(o(i.path ? r(i.path) : p, a));
			if (!t(s)) return "";
			let c = a.split("/").pop() || "demo.vue", f = `Fx_${l(s)}`;
			m(`import ${f} from '${s}'`);
			let g = e.tokens.findIndex((e) => e.content.includes(n));
			return g === -1 || _(g, [c], f, h(c, u(c), `<<< ${s}`, [s])), "";
		});
		for (let t = 0; t < e.tokens.length; t++) {
			let n = e.tokens[t];
			if (n.type !== "html_block") continue;
			let r = n.content.match(/^<DemoPreview>([\s\S]*?)<\/DemoPreview>\s*$/);
			if (!r) continue;
			let i = f(r[1]).trim();
			if (!i) continue;
			let a = i.match(/<script setup>([\s\S]*?)<\/script>/), o = a ? a[1].trim() : "", u = `<template>\n${i.replace(/<script setup>[\s\S]*?<\/script>/, "").trim()}\n</template>` + (o ? `\n<script setup>\n${o}\n<\/script>` : ""), d = l(u);
			c.set(d, u);
			let p = `Fx_${d}`;
			m(`import ${p} from '${s}${d}.vue'`);
			let _ = "demo.vue";
			n.content = `<DemoPreview files="${encodeURIComponent(JSON.stringify([_]))}"><${p}/>`;
			let v = [...h(_, "vue", i), g()];
			e.tokens.splice(t + 1, 0, ...v), t += v.length;
		}
	});
}
function m() {
	return {
		name: "fx-demo-virtual",
		enforce: "pre",
		resolveId(e) {
			if (e.startsWith(s)) return e;
		},
		load(e) {
			if (e.startsWith(s)) {
				let t = e.slice(16).replace(/\.vue$/, "");
				return c.get(t) || "";
			}
		}
	};
}
//#endregion
//#region .vitepress/config/index.ts
var h = null, g = {
	themeConfig: {
		outline: [2, 3],
		musicBall: {
			src: "",
			enable: !0,
			visible: !0,
			autoplay: !1,
			loop: !0
		},
		confetti: !0
	},
	markdown: {
		lineNumbers: !0,
		preConfig(e) {
			e.use(p);
		}
	},
	transformPageData(e) {
		let t = e.frontmatter;
		if (t.layout === "home" && t.hero?.image) {
			let e = t.hero.image;
			h = typeof e == "string" ? {
				light: e,
				dark: e
			} : {
				light: e.light,
				dark: e.dark
			};
		}
		h && (h.light || h.dark) && (e.frontmatter.fxHeroImages = { ...h });
	},
	vite: {
		plugins: [m()],
		ssr: { noExternal: [
			"element-plus",
			"canvas-confetti",
			"node-vibrant",
			"@fuxishi/vitepress-theme"
		] }
	}
};
//#endregion
export { g as default };
