// @fuxishi/vitepress-theme/dist/index.js
import e from "vitepress/theme";
import { Fragment as t, computed as n, createBlock as r, createCommentVNode as i, createElementBlock as a, createElementVNode as o, createSlots as s, createVNode as c, defineComponent as l, h as u, nextTick as d, normalizeClass as f, onMounted as p, onUnmounted as m, openBlock as h, provide as g, ref as _, renderList as v, renderSlot as y, resolveComponent as b, resolveDynamicComponent as x, toDisplayString as S, useSlots as C, vShow as w, withCtx as T, withDirectives as E } from "vue";
import { defineClientComponent as D, useData as O } from "vitepress";
import { ElTooltip as k } from "element-plus";
//#region .vitepress/components/FxLayout.ts
var A = l({ setup(t, { slots: n }) {
	let r = n, { Layout: i } = e, { site: a, isDark: o } = O(), s = a.value.themeConfig;
	g("toggle-appearance", async ({ clientX: e, clientY: t }) => {
		if (!("startViewTransition" in document)) {
			o.value = !o.value;
			return;
		}
		let n = [`circle(0px at ${e}px ${t}px)`, `circle(${Math.hypot(Math.max(e, innerWidth - e), Math.max(t, innerHeight - t))}px at ${e}px ${t}px)`];
		await document.startViewTransition(async () => {
			o.value = !o.value, await d();
		}).ready, document.documentElement.animate({ clipPath: o.value ? [...n].reverse() : n }, {
			duration: 400,
			easing: "ease-in-out",
			fill: "forwards",
			pseudoElement: `::view-transition-${o.value ? "old" : "new"}(root)`
		});
	});
	let c = s.musicBall?.enable && (s.musicBall?.src || s.musicBall?.list?.length > 0), l = _(), f = _(""), h = _(!1), v, y = s.codeBlockFold !== !1, b = typeof s.codeBlockFold == "object" ? s.codeBlockFold.lines : void 0;
	p(() => {
		s.smoothScroll && document.documentElement.classList.add("fx-smooth-scroll"), y && document.documentElement.classList.add("fx-code-fold-enabled");
		let e = () => {
			let e = document.querySelector(".VPNavBarTitle .title span");
			e && (l.value = e, f.value = e.textContent || "", h.value = e.scrollWidth > e.clientWidth);
		}, t = document.querySelector(".VPNavBarTitle .title");
		t && (v = new ResizeObserver(e), v.observe(t)), e();
	}), m(() => v?.disconnect());
	let x = D(() => import("./chunk/FxMusicBall-B2TF7Q7J.js"), [{ style: { display: s.musicBall && s.musicBall.visible ? "" : "none" } }]), S = D(() => import("./chunk/FxHomeFeatureBefore-BHzR4SeT.js")), C = D(() => import("./chunk/FxConfetti-C8bEZxL7.js"), [{ confetti: s.confetti }]), w = D(() => import("./chunk/FxHeroImageBg-BWDDLukW.js"), [{ heroImageColor: s.heroImageColor }]), T = D(() => import("./chunk/FxCodeBlockFold-DGFv2bTG.js"), [{ lines: b }]);
	return () => [
		y && u("style", `:root{--fx-code-fold-lines:${b || 10}}`),
		s.heroImageColor && u("style", ":root{--vp-home-hero-image-background-image:none;--vp-home-hero-name-color:transparent;--vp-home-hero-name-background:none}"),
		u(i, null, {
			...n,
			"home-features-before": () => [u(S), r["home-features-before"]?.()]
		}),
		c && u(x),
		s.confetti && u(C),
		s.heroImageColor && u(w),
		y && u(T),
		l.value && u(k, {
			content: f.value,
			virtualTriggering: !0,
			virtualRef: l.value,
			disabled: !h.value,
			placement: "bottom"
		})
	];
} }), j = { class: "fx-preview-group" }, M = { class: "fx-preview-group__toolbar" }, N = { class: "fx-preview-group__tabs" }, P = ["onClick"], F = ["title"], I = { class: "fx-preview-group__content" }, L = /* @__PURE__ */ l({
	__name: "PreviewGroup",
	props: { files: { default: () => [] } },
	setup(e) {
		let i = e, s = C(), c = _(!1), l = _(i.files[0] || ""), u = n(() => i.files.map((e) => ({
			label: e,
			render: s[e]
		}))), d = (e) => {
			l.value = e, c.value = !0;
		};
		return (e, n) => (h(), a("div", j, [o("div", M, [o("div", N, [(h(!0), a(t, null, v(u.value, (e) => (h(), a("button", {
			key: e.label,
			type: "button",
			class: f(["fx-preview-group__tab", { "is-active": l.value === e.label }]),
			onClick: (t) => d(e.label)
		}, S(e.label), 11, P))), 128))]), o("button", {
			type: "button",
			class: f(["fx-preview-group__toggle", { "is-expanded": c.value }]),
			title: c.value ? "收起源码" : "查看源码",
			onClick: n[0] ||= (e) => c.value = !c.value
		}, [...n[1] ||= [o("svg", {
			viewBox: "0 0 24 24",
			width: "16",
			height: "16",
			"aria-hidden": "true"
		}, [o("path", {
			fill: "none",
			stroke: "currentColor",
			"stroke-width": "2",
			"stroke-linecap": "round",
			"stroke-linejoin": "round",
			d: "m6 9 6 6 6-6"
		})], -1)]], 10, F)]), E(o("div", I, [(h(!0), a(t, null, v(u.value, (e) => E((h(), a("div", {
			key: e.label,
			class: "fx-preview-group__panel"
		}, [(h(), r(x(e.render)))])), [[w, l.value === e.label]])), 128))], 512), [[w, c.value]])]));
	}
}), R = { class: "fx-demo-preview" }, z = { class: "fx-demo-preview__runner not-prose" }, B = /* @__PURE__ */ l({
	__name: "DemoPreview",
	props: { files: { default: "" } },
	setup(e) {
		let t = e, l = n(() => {
			try {
				return JSON.parse(decodeURIComponent(t.files || ""));
			} catch {
				return [];
			}
		});
		return (e, t) => {
			let n = b("ClientOnly");
			return h(), a("div", R, [o("div", z, [c(n, null, {
				fallback: T(() => [...t[0] ||= [o("div", { class: "fx-demo-preview__loading" }, "加载预览中…", -1)]]),
				default: T(() => [y(e.$slots, "default")]),
				_: 3
			})]), l.value.length ? (h(), r(L, {
				key: 0,
				files: l.value
			}, s({ _: 2 }, [v(l.value, (t) => ({
				name: t,
				fn: T(() => [y(e.$slots, t)])
			}))]), 1032, ["files"])) : i("", !0)]);
		};
	}
}), V = {
	extends: e,
	Layout: A,
	enhanceApp({ app: e }) {
		e.component("DemoPreview", B);
	}
};
//#endregion
export { V as default };
