// @fuxishi/vitepress-theme/dist/chunk/FxCodeBlockFold-DGFv2bTG.js
import { t as e } from "./icons-BLv72Cfw.js";
import { defineComponent as t, nextTick as n, onBeforeUnmount as r, onMounted as i, watch as a } from "vue";
import { useRoute as o } from "vitepress";
//#region .vitepress/components/FxCodeBlockFold.ts
var s = "fx-code-folded", c = "fx-code-fold-overlay", l = "div[class*='language-']", u = t({
	props: { lines: {
		type: Number,
		default: 10
	} },
	setup(t) {
		function u() {
			let n = t.lines ?? 10;
			document.querySelectorAll(l).forEach((t) => {
				if (t.classList.contains(c) || t.classList.contains("vp-code-group")) return;
				let r = t.querySelector("pre.shiki code");
				if (!r) return;
				if (r.querySelectorAll(".line").length <= n) {
					t.classList.add("fx-code-no-fold");
					return;
				}
				if (t.querySelector(`.${c}`)) return;
				t.classList.add(s), t.classList.add("fx-code-foldable");
				let i = document.createElement("div");
				i.className = c;
				let a = document.createElement("span");
				a.className = "fx-code-fold-label", a.innerHTML = `展开${e("fold")}`, i.appendChild(a), i.addEventListener("click", () => {
					let n = t.classList.toggle(s);
					i.classList.toggle("fx-code-fold-expanded", !n), a.innerHTML = n ? `展开${e("fold")}` : `收起${e("unfold")}`;
				}), t.appendChild(i);
			});
		}
		let d = null;
		function f() {
			d?.disconnect();
			let e = document.querySelector(".vp-doc");
			e && (d = new MutationObserver(() => {
				requestAnimationFrame(u);
			}), d.observe(e, {
				childList: !0,
				subtree: !0
			}));
		}
		i(() => {
			u(), f();
		});
		let p = o();
		return a(() => p.path, () => {
			n(() => {
				u(), f();
			});
		}), r(() => {
			d?.disconnect();
		}), () => null;
	}
});
//#endregion
export { u as default };
