// @fuxishi/vitepress-theme/dist/chunk/FxConfetti-C8bEZxL7.js
import { defineComponent as e, onBeforeUnmount as t, onMounted as n } from "vue";
import r from "canvas-confetti";
//#endregion
//#region .vitepress/components/FxConfetti.vue
var i = /* @__PURE__ */ e({
	__name: "FxConfetti",
	props: { confetti: { type: [Boolean, Object] } },
	setup(e) {
		let i = e, a = typeof i.confetti == "object" ? i.confetti : {}, o = a.shape || "star", s = a.secondary !== !1, c = a.shapes ? Array.isArray(a.shapes) ? a.shapes : [a.shapes] : null, l = [], u = [];
		function d(e, t, n, i = "circle") {
			let a = () => {
				r({
					...n,
					particleCount: 10,
					scalar: 1.2
				}), r({
					...n,
					particleCount: 10,
					scalar: .75,
					...s ? { shapes: [i] } : {}
				});
			};
			u.push(setTimeout(a, 0)), u.push(setTimeout(a, 50)), u.push(setTimeout(a, 100));
		}
		let f = (e, t) => d(e, t, {
			spread: 360,
			ticks: 50,
			gravity: 0,
			decay: .94,
			startVelocity: 10,
			colors: [
				"FFE400",
				"FFBD00",
				"E89400",
				"FFCA6C",
				"FDFFB8"
			],
			origin: {
				x: e,
				y: t
			},
			shapes: ["star"]
		}), p = (e, t) => d(e, t, {
			spread: 360,
			ticks: 50,
			gravity: 0,
			decay: .94,
			startVelocity: 10,
			colors: [
				"#FF1461",
				"#18FF92",
				"#5A87FF",
				"#FBF38C",
				"#FF6B35"
			],
			origin: {
				x: e,
				y: t
			}
		}, "square"), m = (e, t) => d(e, t, {
			spread: 360,
			ticks: 50,
			gravity: 0,
			decay: .94,
			startVelocity: 10,
			shapes: l,
			origin: {
				x: e,
				y: t
			}
		});
		function h(e) {
			let t = e.clientX / window.innerWidth, n = e.clientY / window.innerHeight;
			l.length > 0 ? m(t, n) : o === "colored-paper" ? p(t, n) : f(t, n);
		}
		return n(() => {
			c && (l = c.map((e) => r.shapeFromText({
				text: e,
				scalar: 2
			}))), window.addEventListener("click", h);
		}), t(() => {
			window.removeEventListener("click", h), u.forEach(clearTimeout);
		}), (e, t) => null;
	}
});
//#endregion
export { i as default };
