// @fuxishi/vitepress-theme/dist/chunk/FxHeroImageBg-BWDDLukW.js
import { defineComponent as e, nextTick as t, onBeforeUnmount as n, onMounted as r, watch as i } from "vue";
import { useData as a, useRoute as o, withBase as s } from "vitepress";
import { Vibrant as c } from "node-vibrant/browser";
//#region .vitepress/components/FxHeroImageBg.ts
var l = "fx-hero-colors";
function u() {
	try {
		let e = localStorage.getItem(l);
		return e ? JSON.parse(e) : null;
	} catch {
		return null;
	}
}
function d(e) {
	try {
		localStorage.setItem(l, JSON.stringify(e));
	} catch {}
}
function f(e) {
	return [
		e.Vibrant,
		e.DarkVibrant,
		e.LightVibrant,
		e.Muted,
		e.DarkMuted,
		e.LightMuted
	].filter(Boolean);
}
function p(e) {
	return e.slice().sort((e, t) => e.hsl[0] - t.hsl[0]).map((e) => e.hex);
}
function m(e) {
	let t = f(e);
	if (t.length === 0) return "-webkit-linear-gradient(-45deg, #bd34fe 50%, #47caff 50%)";
	let n = p(t);
	return `conic-gradient(from 0deg, ${[...n, n[0]].join(", ")})`;
}
function h(e, t) {
	let n = Math.abs(e - t);
	return Math.min(n, 360 - n);
}
function g(e) {
	let t = f(e);
	if (t.length < 2) return "linear-gradient(120deg, #bd34fe, #41d1ff)";
	let n = -1, r = [t[0].hex, t[1].hex];
	for (let e = 0; e < t.length; e++) for (let i = e + 1; i < t.length; i++) {
		let a = h(t[e].hsl[0], t[i].hsl[0]);
		a > n && (n = a, r = [t[e].hex, t[i].hex]);
	}
	return `linear-gradient(120deg, ${r[0]}, ${r[1]})`;
}
function _(e) {
	return `${parseInt(e.slice(1, 3), 16)}, ${parseInt(e.slice(3, 5), 16)}, ${parseInt(e.slice(5, 7), 16)}`;
}
function v(e) {
	let t = f(e);
	if (t.length < 2) return ["189, 52, 254", "65, 209, 255"];
	let n = -1, r = [t[0], t[1]];
	for (let e = 0; e < t.length; e++) for (let i = e + 1; i < t.length; i++) {
		let a = h(t[e].hsl[0], t[i].hsl[0]);
		a > n && (n = a, r = [t[e], t[i]]);
	}
	return [_(r[0].hex), _(r[1].hex)];
}
function y() {
	return document.documentElement.classList.contains("dark");
}
function b(e) {
	let [t, n] = v(e);
	return {
		imageGradient: m(e),
		nameGradient: g(e),
		beamC1: t,
		beamC2: n
	};
}
function x(e) {
	let t = document.documentElement;
	t.style.setProperty("--vp-home-hero-image-background-image", e.imageGradient), t.style.setProperty("--vp-home-hero-name-background", e.nameGradient), t.style.setProperty("--fx-beam-c1", e.beamC1), t.style.setProperty("--fx-beam-c2", e.beamC2);
}
function S(e) {
	let t = document.documentElement;
	t.style.setProperty("--fx-beam-c1", e.beamC1), t.style.setProperty("--fx-beam-c2", e.beamC2);
}
var C = e({
	props: { heroImageColor: {
		type: Boolean,
		default: !1
	} },
	render: () => null,
	setup(e) {
		if (!e.heroImageColor) return;
		let f = null, p = o(), { frontmatter: m, site: h } = a();
		function g() {
			let e = document.documentElement;
			e.style.removeProperty("--vp-home-hero-image-background-image"), e.style.removeProperty("--vp-home-hero-name-background"), e.style.removeProperty("--fx-beam-c1"), e.style.removeProperty("--fx-beam-c2");
		}
		function _(e) {
			return y() ? e.dark || e.light : e.light || e.dark;
		}
		function v(e) {
			let t = u();
			return !t || !t.images ? !1 : e && (e.light && t.images.light && e.light !== t.images.light || e.dark && t.images.dark && e.dark !== t.images.dark) ? (localStorage.removeItem(l), !0) : !1;
		}
		function C() {
			let e = u();
			if (!e) return !1;
			let t = _(e);
			return t ? (x(t), !0) : !1;
		}
		function w() {
			let e = u();
			if (!e) return !1;
			let t = _(e);
			return t ? (S(t), !0) : !1;
		}
		function T() {
			return m.value.fxHeroImages || {};
		}
		async function E(e, t) {
			let n = e.src;
			if (n) try {
				let e = b(await c.from(n).quality(1).getPalette());
				(y() && t === "dark" || !y() && t === "light") && x(e);
				let r = u() || {};
				r[t] = e, r.images ||= {}, r.images[t] = n, d(r);
			} catch {}
		}
		async function D(e, t) {
			try {
				let n = b(await c.from(e).quality(1).getPalette()), r = u() || {};
				return r[t] = n, r.images ||= {}, r.images[t] = e, d(r), n;
			} catch {
				return null;
			}
		}
		async function O(e, t = !0) {
			if (!e.light && !e.dark) return;
			let n = {};
			if (e.light) {
				let r = await D(t ? s(e.light) : e.light, "light");
				r && (n.light = r);
			}
			if (e.dark) {
				let r = await D(t ? s(e.dark) : e.dark, "dark");
				r && (n.dark = r);
			}
			if (n.light || n.dark) {
				let e = y() ? n.dark || n.light : n.light || n.dark;
				e && x(e);
			}
		}
		async function k() {
			let e = T();
			if (e.light || e.dark) {
				await O(e);
				return;
			}
			try {
				let e = await (await fetch(window.location.origin + h.value.base)).text(), t = new DOMParser().parseFromString(e, "text/html").querySelectorAll(".VPHero .image-container img.VPImage"), n = {};
				t.forEach((e) => {
					let t = e.getAttribute("src");
					t && (e.classList.contains("dark") ? n.dark = t : n.light = t);
				}), (n.light || n.dark) && await O(n, !1);
			} catch {}
		}
		function A() {
			let e = document.querySelector(".VPHero .image-container img.VPImage.light"), t = document.querySelector(".VPHero .image-container img.VPImage.dark");
			e || t ? (e && e.complete && e.naturalWidth > 0 ? E(e, "light") : e && e.addEventListener("load", () => E(e, "light"), { once: !0 }), t && t.complete && t.naturalWidth > 0 ? E(t, "dark") : t && t.addEventListener("load", () => E(t, "dark"), { once: !0 })) : (g(), w(), k());
		}
		r(() => {
			v(T()), C() || k(), A(), f = new MutationObserver((e) => {
				for (let t of e) if (t.attributeName === "class") {
					w(), A();
					break;
				}
			}), f.observe(document.documentElement, {
				attributes: !0,
				attributeFilter: ["class"]
			});
		}), i(() => p.path, () => {
			t(A);
		}), n(() => {
			f?.disconnect(), g();
		});
	}
});
//#endregion
export { C as default };
