// @fuxishi/vitepress-theme/dist/chunk/FxMusicBall-B2TF7Q7J.js
import { t as e } from "./icons-BLv72Cfw.js";
import { Fragment as t, computed as n, createCommentVNode as r, createElementBlock as i, createElementVNode as a, createVNode as o, defineComponent as s, nextTick as ee, normalizeClass as c, onBeforeUnmount as l, onMounted as u, openBlock as d, ref as f, renderList as p, toDisplayString as m, unref as h, vShow as g, watchEffect as _, withDirectives as v } from "vue";
import { useData as y, withBase as b } from "vitepress";
import { ElSlider as x } from "element-plus";
//#region .vitepress/components/FxMusicBall.vue?vue&type=script&setup=true&lang.ts
var S = [
	"src",
	"autoplay",
	"loop"
], C = ["src", "autoplay"], w = { class: "fx-music-ball__inner" }, T = { class: "fx-play-pause" }, E = ["innerHTML"], D = ["innerHTML"], O = { class: "fx-time" }, k = { class: "fx-duration" }, te = { class: "fx-volume" }, ne = ["innerHTML"], re = ["innerHTML"], ie = ["innerHTML"], ae = {
	id: "fx-progress-gradient",
	x1: "0%",
	y1: "0%",
	x2: "100%",
	y2: "100%"
}, oe = ["stop-color"], se = ["stop-color"], ce = ["stroke-dasharray"], le = {
	key: 2,
	class: "fx-orbit-ring"
}, ue = {
	key: 3,
	class: "fx-music-list"
}, de = ["onClick"], fe = ["d"], A = /* @__PURE__ */ s({
	__name: "FxMusicBall",
	setup(s) {
		let A = [
			"M123.252 73.826h111.169v790.935h-111.169z",
			"M346.361 465.952h111.169v395.467h-111.169z",
			"M569.466 229.325h111.169v636.444h-111.169z",
			"M792.574 391.584h111.169v475.788h-111.169z"
		], { site: j } = y(), M = j.value.themeConfig.musicBall, N = f(null), P = f(0), F = f(0), I = f(0), L = f(!1), R = f(50), z = f(!1), B = f(0), V = f(!!M.autoplay), H = f(!1), U = 2 * Math.PI * 22, W = f("#bd34fe"), G = f("#41d1ff");
		function K() {
			let e = getComputedStyle(document.documentElement), t = e.getPropertyValue("--fx-beam-c1").trim(), n = e.getPropertyValue("--fx-beam-c2").trim();
			t && (W.value = `rgb(${t})`), n && (G.value = `rgb(${n})`);
		}
		let q = null;
		u(() => {
			K(), q = new MutationObserver((e) => {
				for (let t of e) if (t.attributeName === "style") {
					K();
					break;
				}
			}), q.observe(document.documentElement, {
				attributes: !0,
				attributeFilter: ["style"]
			});
		}), l(() => {
			q?.disconnect();
		});
		let pe = n(() => {
			let e = U - X.value / 100 * U;
			return `${U - e} ${e}`;
		}), J = n(() => M.src || M.list?.length === 1), me = n(() => b(M.src || M.list?.[0]?.src || "")), Y = n(() => (M.list || []).map((e) => ({
			...e,
			src: b(e.src)
		}))), X = n({
			get() {
				return I.value / P.value * 100 || 0;
			},
			set(e) {
				F.value = P.value * e / 100, !L.value && N.value && (N.value.currentTime = F.value);
			}
		}), he = n(() => $(P.value)), Z = n(() => $(I.value));
		_(() => {
			N.value && (z.value = !1, N.value.volume = R.value / 100);
		}), _(() => {
			N.value && (N.value.muted = z.value);
		});
		let Q = {};
		u(() => {
			let e = N.value;
			if (!e) return;
			let t = {
				play: () => {
					L.value = !0;
				},
				pause: () => {
					L.value = !1;
				},
				timeupdate: () => {
					P.value = e.duration, I.value = e.currentTime;
				},
				canplay: () => {
					P.value = e.duration;
				},
				ended: () => {
					if (B.value + 1 > Y.value.length - 1) {
						if (!M.loop) {
							L.value = !1;
							return;
						}
						B.value = 0;
					} else B.value++;
					ee(() => e.play());
				}
			};
			for (let [n, r] of Object.entries(t)) e.addEventListener(n, r);
			Q = t;
		}), l(() => {
			let e = N.value;
			if (e) {
				for (let [t, n] of Object.entries(Q)) e.removeEventListener(t, n);
				Q = {};
			}
		});
		let ge = () => {
			L.value = !L.value, L.value ? N.value?.play() : N.value?.pause(), !J.value && !V.value && (V.value = !0);
		}, _e = () => {
			N.value && (N.value.currentTime = F.value);
		}, ve = (e) => {
			V.value = !0, B.value = e;
		}, ye = () => {
			H.value = !1;
		};
		function $(e) {
			return e = Math.max(0, Number(e) || 0), `${Math.floor(e % 3600 / 60)}:${Math.floor(e % 60).toString().padStart(2, "0")}`;
		}
		return (n, s) => (d(), i("div", {
			class: c(["fx-music-ball", {
				active: H.value,
				"is-playing": L.value
			}]),
			onMouseleave: ye
		}, [
			J.value ? (d(), i("audio", {
				key: 0,
				class: "audio",
				src: me.value,
				autoplay: h(M).autoplay,
				loop: h(M).loop,
				style: { display: "none" },
				ref_key: "audioRef",
				ref: N
			}, null, 8, S)) : (d(), i("audio", {
				key: 1,
				class: "audio",
				src: Y.value[B.value]?.src,
				autoplay: V.value,
				style: { display: "none" },
				ref_key: "audioRef",
				ref: N
			}, null, 8, C)),
			a("div", w, [
				a("div", T, [v(a("span", { innerHTML: h(e)("music-ball/playfill", 20) }, null, 8, E), [[g, !L.value]]), v(a("span", { innerHTML: h(e)("music-ball/pause", 26) }, null, 8, D), [[g, L.value]])]),
				a("div", O, m(Z.value) + "\xA0/\xA0" + m(he.value), 1),
				a("div", k, [o(h(x), {
					style: { width: "100%" },
					"show-tooltip": !1,
					modelValue: X.value,
					"onUpdate:modelValue": s[0] ||= (e) => X.value = e,
					onChange: _e
				}, null, 8, ["modelValue"])]),
				a("div", te, [
					o(h(x), {
						style: { width: "100%" },
						"show-tooltip": !1,
						modelValue: R.value,
						"onUpdate:modelValue": s[1] ||= (e) => R.value = e
					}, null, 8, ["modelValue"]),
					v(a("span", {
						class: "fx-icon-volume",
						onClick: s[2] ||= (e) => z.value = !z.value,
						innerHTML: h(e)("music-ball/volume_off", 20)
					}, null, 8, ne), [[g, z.value]]),
					v(a("span", {
						class: "fx-icon-volume",
						onClick: s[3] ||= (e) => z.value = !z.value,
						innerHTML: h(e)("music-ball/volume_up", 20)
					}, null, 8, re), [[g, !z.value]])
				]),
				J.value ? r("", !0) : (d(), i("div", {
					key: 0,
					class: "fx-music-list-button",
					onClick: s[4] ||= (e) => H.value = !H.value
				}, [a("span", { innerHTML: h(e)("music-ball/music_list", 20) }, null, 8, ie)]))
			]),
			(d(), i("svg", {
				class: "fx-progress-ring",
				width: "50",
				height: "50",
				viewBox: "0 0 50 50",
				onClick: ge
			}, [
				a("defs", null, [a("linearGradient", ae, [a("stop", {
					offset: "0%",
					"stop-color": W.value
				}, null, 8, oe), a("stop", {
					offset: "100%",
					"stop-color": G.value
				}, null, 8, se)])]),
				s[5] ||= a("circle", {
					class: "fx-progress-track",
					cx: "25",
					cy: "25",
					r: "22",
					fill: "none",
					"stroke-width": "2.5"
				}, null, -1),
				a("circle", {
					class: "fx-progress-bar",
					cx: "25",
					cy: "25",
					r: "22",
					fill: "none",
					stroke: "url(#fx-progress-gradient)",
					"stroke-width": "2.5",
					"stroke-linecap": "round",
					"stroke-dasharray": pe.value
				}, null, 8, ce)
			])),
			L.value ? (d(), i("div", le)) : r("", !0),
			H.value ? (d(), i("div", ue, [(d(!0), i(t, null, p(Y.value, (e, n) => (d(), i("div", {
				class: c(["fx-music-list-item", B.value === n ? "active" : ""]),
				key: n,
				onClick: (e) => ve(n)
			}, [B.value === n ? (d(), i("svg", {
				key: 0,
				class: c(["image-live", { playing: L.value }]),
				viewBox: "0 0 1024 1024",
				xmlns: "http://www.w3.org/2000/svg"
			}, [(d(), i(t, null, p(A, (e, t) => a("path", {
				key: t,
				d: e,
				class: c(["bar", `bar-${t + 1}`])
			}, null, 10, fe)), 64))], 2)) : r("", !0), a("span", null, m(e.name), 1)], 10, de))), 128)), s[6] ||= a("div", { class: "fx-music-list-sep" }, null, -1)])) : r("", !0)
		], 34));
	}
});
//#endregion
export { A as default };
